spec/controllers/all_casa_admins/casa_admins_controller_spec.rb

```diff
@@ -0,0 +1,170 @@
+require "rails_helper"
+
+RSpec.describe AllCasaAdmins::CasaAdminsController, type: :controller do
+  let(:all_casa_admin) { build(:all_casa_admin) }
```

Comment:
the id is nil because you are using build not create. It would be a nicer style to use create and not manually set the id.

Reply:
Changing it to create fixed the issue. Thanks for pointing this out.